File: in_Java/1395 Count Number of Teams.java
class Solution {
    public int numTeams(int[] rating) {
        int res = 0;
        if(rating.length < 3) return res;
        for(int j = 0; j < rating.length-1; j++){
            int leftSmaller = 0, leftLarger = 0;
            int rightSmaller = 0, rightLarger = 0;
            for(int i = 0; i < j; i++){
                if(rating[i] < rating[j]) leftSmaller++;
                if(rating[i] > rating[j]) leftLarger++;
            }
            for(int k = j+1; k < rating.length; k++){
                if(rating[k] > rating[j]) rightLarger++;
                if(rating[k] < rating[j]) rightSmaller++;
            }
            res += leftSmaller * rightLarger + leftLarger * rightSmaller;
        }
        return res;
    }
}

File: in_Python/1395 Count Number of Teams.py
class Solution(object):
    def numTeams(self, rating):
        """
        :type rating: List[int]
        :rtype: int
        """
        res = 0
        l = len(rating)
        for i in range(l):
            for j in range(i+1, l):
                for k in range(j+1, l):
                    if rating[i] < rating[j] < rating[k] or rating[i] > rating[j] > rating[k]:
                        res += 1
        return res